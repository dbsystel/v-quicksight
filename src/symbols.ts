export const EmbeddingContextInjectionKey = Symbol(
  "The key to be used with inject when accessing the created embeddingContext for usage with the amazon-quicksight-embedding-sdk",
)
