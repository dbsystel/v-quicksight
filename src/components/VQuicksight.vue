<script setup lang="ts">
import { createEmbeddingContext } from 'amazon-quicksight-embedding-sdk'
import type {
  EmbeddingContext,
  ExperienceFrameMetadata,
  SimpleChangeEvent
} from 'amazon-quicksight-embedding-sdk/dist/types'
import { onBeforeMount, provide, ref } from 'vue'
import { EmbeddingContextInjectionKey } from '../symbols'

withDefaults(
  defineProps<{
    className?: string
  }>(),
  {
    className: 'v-quicksight'
  }
)

const emit = defineEmits<{
  (e: 'change', data: { changeEvent: SimpleChangeEvent; metadata?: ExperienceFrameMetadata }): void
}>()

const embeddingContext = ref<EmbeddingContext>()

onBeforeMount(setupEmbeddingContext)

async function setupEmbeddingContext() {
  embeddingContext.value = await createEmbeddingContext({
    onChange: (changeEvent, metadata) => {
      emit('change', { changeEvent, metadata })
    }
  })
}

provide(EmbeddingContextInjectionKey, embeddingContext)
</script>

<template>
  <div :class="className"><slot></slot></div>
</template>

<style>
.quicksight-embedding-iframe {
  height: 100%;
}
</style>
