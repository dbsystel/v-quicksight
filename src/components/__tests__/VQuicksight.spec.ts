import VQuicksight from '../VQuicksight.vue'
import { EmbeddingContextInjectionKey } from '../../symbols'
import { mount, shallowMount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import { inject, nextTick } from 'vue'

describe('VQuicksight', () => {
  it('renders as expected with default CSS class', () => {
    const component = shallowMount(VQuicksight)
    expect(component.element).toMatchSnapshot()
  })

  it('renders as expected with custom CSS class', () => {
    const component = shallowMount(VQuicksight, { props: { className: 'my-class' } })
    expect(component.element).toMatchSnapshot()
  })

  describe('Integration tests', () => {
    function setupComponents() {
      const ChildComponent = {
        template: '',
        setup() {
          return {
            ctx: inject(EmbeddingContextInjectionKey)
          }
        }
      }
      const component = mount(VQuicksight, {
        slots: {
          default: ChildComponent
        }
      })
      const child = component.findComponent(ChildComponent)
      return { component, child }
    }

    it('provides an embeddingContext which can be injected in a child component', async () => {
      const { child } = setupComponents()
      expect(child.vm.ctx).toBeUndefined()
      await nextTick()
      expect(child.vm.ctx).toEqual(
        expect.objectContaining({
          embedConsole: expect.anything(),
          embedDashboard: expect.anything(),
          embedQSearchBar: expect.anything(),
          embedVisual: expect.anything()
        })
      )
    })
  })
})
