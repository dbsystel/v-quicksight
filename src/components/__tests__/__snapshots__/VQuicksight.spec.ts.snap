// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VQuicksight > renders as expected with custom CSS class 1`] = `
<div
  class="my-class"
>
  
  
</div>
`;

exports[`VQuicksight > renders as expected with default CSS class 1`] = `
<div
  class="v-quicksight"
>
  
  
</div>
`;
